src/main/java/net/backupcup/mcde/screen/GildingFoundryScreen.java

```diff
@@ -112,7 +114,7 @@ private boolean isGildingButtonClickable() {
             (inventory.getStack(1).getCount() >= MCDEnchantments.getConfig().getGildingCost() || player.isCreative()) &&
             !handler.hasProgress() &&
             !(slots != null && slots.hasGilding()) &&
-            EnchantmentUtils.canGenerateEnchantment(weapon);
+            handler.hasNewEnchantment();
```

Comment:
The name change from `canGenerateEnchantment` to `hasNewEnchantment`.
Whilst I can see that the `EnchantmentUtils` are no longer being used, what is the reason for the name `hasNewEnchantment`? As I read this, the name does not clearly explain what the check is for. Does it consider innate enchantments? Is it checking to see if it is able to add a new enchantment? There are too many questions that are presented by this name change. Therefore, please clarify via a comment or consider renaming the method.

Reply:
Generation of enchantment now moved to `onSlotUpdate` because now I have to send something to a client to indicate that gilding can be done. So I send an `Optional<Identifier>` to indicate this. `hasNewEnchantment` means that an enchantment for a gilding has been generated, allowing player to press the button.

Maybe I can rename this to `hasEnchantmentForGilding` for more clarity.

Reply:
I think that your suggestion is a very good idea and does clarify.